apps/server/src/modules/pseudonym/uc/pseudonym.uc.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { Test, TestingModule } from '@nestjs/testing';
+import { EntityId, LegacySchoolDo, Pseudonym, SchoolEntity, User } from '@shared/domain';
+import { legacySchoolDoFactory, pseudonymFactory, schoolFactory, setupEntities, userFactory } from '@shared/testing';
+import { ForbiddenException } from '@nestjs/common';
+import { ICurrentUser } from '@src/modules/authentication';
+import { Action, AuthorizationService } from '@src/modules/authorization';
+import { PseudonymService } from '../service';
+import { PseudonymUc } from './pseudonym.uc';
+import { LegacySchoolService } from '../../legacy-school';
+
+describe('PseudonymUc', () => {
+	let module: TestingModule;
+	let uc: PseudonymUc;
+
+	let pseudonymService: DeepMocked<PseudonymService>;
+	let authorizationService: DeepMocked<AuthorizationService>;
+	let schoolService: DeepMocked<LegacySchoolService>;
+
+	beforeAll(async () => {
+		await setupEntities();
+
+		module = await Test.createTestingModule({
+			providers: [
+				PseudonymUc,
+				{
+					provide: PseudonymService,
+					useValue: createMock<PseudonymService>(),
+				},
+				{
+					provide: AuthorizationService,
+					useValue: createMock<AuthorizationService>(),
+				},
+				{
+					provide: LegacySchoolService,
+					useValue: createMock<LegacySchoolService>(),
+				},
+			],
+		}).compile();
+
+		uc = module.get(PseudonymUc);
+		pseudonymService = module.get(PseudonymService);
+		authorizationService = module.get(AuthorizationService);
+		schoolService = module.get(LegacySchoolService);
+	});
+
+	beforeEach(() => {
+		jest.resetAllMocks();
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('findPseudonymByPseudonym', () => {
+		describe('when valid user and params are given', () => {
+			const setup = () => {
+				const currentUser: ICurrentUser = { userId: 'userId' } as ICurrentUser;
```

Comment:
I think you can remove the ICurrentUser here since you don't use it anymore

Reply:
all done